cls/SourceControl/Git/Utils.cls

```diff
@@ -322,6 +334,22 @@ ClassMethod SwitchBranch(targetBranchName As %String) As %Status
     quit $$$OK
 }
 
+ClassMethod Sync() As %Status
+{
+    write "Syncing local repository..."
+    if '..HasRemoteRepo() {
+        write "No remote repository configured: skipping fetch, pull and push"
+        do ..Commit("", "Committing all files through sync command")
```

Comment:
This should:
* Build a list of files with tracked changes (staged/unstaged, we want all of them)
* Build a reasonable commit message based on that
* Offer the user the opportunity to customize the commit message (via a different "Action")

The flow for this is going to be building out the list, commit message, offering it to the user in UserAction and then actually doing the Fetch/Pull/Commit/Push in AfterUserAction (from the extension itself - might still want to have helper methods in Utils for consistency, e.g. PreSync and DoSync)